Sources/StreamChatUI/CommonViews/AvatarView/ChatChannelAvatarView.swift

```diff
@@ -33,28 +33,204 @@ open class ChatChannelAvatarView: _View, ThemeProvider, SwiftUIRepresentable {
+                from: channelAvatarUrl,
+                placeholder: appearance.images.userAvatarPlaceholder4,
+                preferredSize: .avatarThumbnailSize,
+                components: components
+            )
+            return
+        }
+        
+        let lastActiveMembers = channel.lastActiveMembers
+            .sorted { $0.memberCreatedAt < $1.memberCreatedAt }
+            .filter { $0.id != content.currentUserId }
+        
+        // If there are no members other than the current user in the channel, load a placeholder
+        if lastActiveMembers.isEmpty {
+            presenceAvatarView.avatarView.imageView.loadImage(
+                from: nil,
+                placeholder: appearance.images.userAvatarPlaceholder4,
+                preferredSize: .avatarThumbnailSize,
+                components: components
+            )
+            return
+        }
+        
+        // If the channel is a direct message channel, load the other user's avatar
+        // and also set the online indicator visible
+        if channel.isDirectMessageChannel {
+            let otherMember = lastActiveMembers.first
+            let isOnlineIndicatorVisible = otherMember?.isOnline ?? false
+            
+            presenceAvatarView.avatarView.imageView.loadImage(
+                from: otherMember?.imageURL,
+                placeholder: appearance.images.userAvatarPlaceholder3,
+                preferredSize: .avatarThumbnailSize,
+                components: components
+            )
+            
+            presenceAvatarView.isOnlineIndicatorVisible = isOnlineIndicatorVisible
+            return
+        }
+        
+        // The channel is a non-DM channel, hide the online indicator
+        presenceAvatarView.isOnlineIndicatorVisible = false
+        
+        var urls = lastActiveMembers.map(\.imageURL)
+        
+        if urls.isEmpty {
+            presenceAvatarView.avatarView.imageView.loadImage(
+                from: nil,
+                placeholder: appearance.images.userAvatarPlaceholder3,
+                preferredSize: .avatarThumbnailSize,
+                components: components
+            )
+            return
+        } else if urls.count > 4 {
+            // We show a combination of at max 4 images combined
+            urls = Array(urls[0...3])
+        }
+        
+        guard !SystemEnvironment.isTests else {
+            // When running tests, we load the images synchronously
+            let images = urls.map { UIImage(data: try! Data(contentsOf: $0!))! }
+           
+            let combinedImage = getCombinedAvatarImage(from: images) ?? appearance.images.userAvatarPlaceholder2
+            
+            presenceAvatarView.avatarView.imageView.loadImage(
+                from: nil,
+                placeholder: combinedImage,
+                preferredSize: .avatarThumbnailSize,
+                components: components
+            )
+            return
+        }
```

Comment:
It isn't a good sign we have `SystemEnvironment.isTests` everywhere. Ideally, we should replace the ´ImageLoader` implementation in the tests, with a mocked one. The problem here is that `loadImage` is an extension on `UIImageView`, but I would prefer that we have an `ImageLoader` component that is responsible to load all the images and handle all of that. This loader should be a dependency of each component that needs to load images. So it is nice that you introduced `ImageLoader`, but I would not make it a singleton, and would make it a dependency of each component that needs it so that it is easily testable + easier to change implementations. And so, I would also move the `UIImageView.loadImage` extension to `ImageLoader`.

Reply:
yes, agree this completely breaks SOLID and makes code tight-coupled

Reply:
Removed the tests check, and also removed the `UIImageView` extension.